### ISSUE TITLE
SA Hosted billing cycle issues

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#550]
In GitLab by @kdjstudios on Nov 3, 2017, 15:48

**Submitted by:**  "Michelle McKee" <michelle.mckee@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2017-11-03-79854

**Server:** Internal

**Client/Site:** SA Hosted

**Account:** All

**Issue:** 

I just went in today to try to do the 11/1/17 billing cycle for SA Hosted and Management Services and I am running into trouble.
 

1)      I tried to put staged fee in customer SAB-1124 but the note on the account states that “Staged fees and payments will be applied in 12/1/17 Master billing cycle.”  I don’t understand why.  I have not tried to run the 11/1/17 billing cycle before today.

2)      I went into the billing cycles menu and tried to change dates to do the 11/1/17 billing cycle and a whole new menu popped up.  Is this the new way?
 
![image002](/uploads/74d874598fc0075ee35fad92aff938e0/image002.jpg)

3)      When I got around to the apply portion of the billing.  I got the below “Pending Invoices” screen.  I have never seen this before.
 
![image007](/uploads/7e6e67f4bfe8395f3b937e5e0426428c/image007.jpg)

 

4)      I reverted the billing cycle that I had completed so far.  Then went to check the different customers to see if anything was weird before I proceeded.  I hit two customers that no matter what I did I get the below message
 
![image006](/uploads/6ef7b01f15f88b0c8018e8c25ff3c053/image006.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 3, 2017, 15:48

My Reply back:


We will detail out responses below to each individual topic you submitted.

1. Staged fees will always go to the next cycle from the date specified in the site edit page as"last billing cycle". This is most likely caused if there was ever a billing cycle created for the 11/1/17 cycle, even if it was reverted. To correct this, you will need to change the 'last billing cycle' to the date of the correct last cycle. You will also need to revert the billing cycle, if you are in the process of one. Example: If you will want to set the last billing cycle to 10/1/17; so then the next cycle to be the 11/1/17. if you add your staged fees now the notice should say that the staged fees will be applied to the 11/1/17 cycle. Once done adding staged fees go back to the billing cycle page and create the next cycle with the dates of 11/1/17.  Also all staged fees must be added before you click the 'apply' step when creating a billing cycle. 

2. To change the billing cycle date you need to alter the 'last billing cycle' date in the site edit page as stated above. Changing the dates here, will only all to change the dates that are printed on the invoices.

3. If there are draft invoices on accounts these must either first be finalized or deleted before you can 'apply' or create a new draft invoice. This is due to the system only allowing one draft invoice per account at a time. To access draft invoices simply click on the 'draft invoice' button. From there you may choose the red "X" in the corner to delete the invoice, or you can choose submit to finalize the invoice.

4. We have informed the support team how the steps you have taken and the error you are receiving.  Would you be able to provide us with an example account/customer please.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 3, 2017, 15:49

@smasih @tim.traylor @kace.coady 

I believe the first three topics I was able to provide solutions for. The forth where the site is receiving the error please look into.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 6, 2017, 24:58

@kdjstudios 

> To change the billing cycle date you need to alter the 'last billing cycle' date in the site edit page as stated above. Changing the dates here, will only all to change the dates that are printed on the invoices.

Invoice Date is real, however usage and service dates are not real

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 6, 2017, 01:16

@kdjstudios 

there are no activities on ~~account~~ site level but still there are few selected on account level. 

This site has a manual switch so why there were requirement of selecting an activity on account level.

The only way out is to delete all the account activities those are selected on account for that site and which would be right as well because site has a manual switch.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 6, 2017, 15:06

@smasih 

Reply Update:

Thank you, Kyle for your help.  Just some additional notes/answers.

1)I am not sure how the last billing cycle got changed.  I am the only one, as far as I know, that handles the billing for the SA Hosted accounts and the last billing I ran was for 10/1/17.  Not a huge deal as I was able to go into the site and change the "last billing cycle" date.

2)My original comment here was that I was getting different screens when I went to do my billing cycle.  Again, no big deal I was just wondering if the screens had recently been changed.

3)How would draft invoices be created?  Who would create them?  As far as I know, I am the only one who handles the billing for the SA Hosted accounts.

4) and now to answer your question....one of the customers that I am getting the "We're sorry, but something went wrong." message is  HastingsHumans under the SA Hosted site.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 6, 2017, 15:12

My reply:


We are glad to hear that adjusting the last billing cycle date fixed the first topic. In response to your second topic, yes you are correct here was a change within the last month on the "change dates" screen. Your third topic; most operational management employees that have access to the site can create draft invoices, due to this; an email is sent out when a manually create draft invoices is created, finalized, or deleted. These emails are sent to the email address specified under the site settings page I believe. In regards to how they are created; a manual draft invoice can be created at anytime from the account page; there is a button labeled "create invoice". Lastly for your forth topic; we have informed the support team of the customer in question and will follow up with their response accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 6, 2017, 23:22

@kdjstudios 

Sorry for my last comment which was very confusing. So, I have edited my comment hope it will make some sense now.

The only way to fix customer accounts is by deleting all the activities within all the accounts and we got to do it from the backend.

**In order to receive email for Invoice drafting, editing and finalization we need to select a site manager  in [admin](https://answernet.sabilling.com/config_emails/config_email_deactivation) section.**  
**If no one is selected then by default notification email will go to `apperrors@sahosted.com`**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 7, 2017, 13:00

@smasih 

Thank you for the clarifications. In regards to the draft emails can we add Michelle as an option on there for the SA Hosted Site please? @tim.traylor Or should these only go to Rich?

@tim.traylor @kace.coady 

With your approval I would say we need to delete all activities from within those accounts effected. 

Also should we be looking into finding a way to prevent this from happening in the future?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 7, 2017, 13:58

Michelle is wondering how the customer accounts become corrupted. Was it something a user change caused, or was this a technical issue? If it was a user change that caused this, She would like to know which user if possible. 

I initially replied, that to our current understanding is that this was not caused by a user change, but was caused by a technical issue and that we will follow up with an update accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 7, 2017, 23:19

@kdjstudios 

SA Hosted has a manual switch so under ticket #492  we removed all the non vcc activities from the non vcc sites.  

However, pre-existed vcc activities were selected in SA hosted accounts which shouldn't have been there in the first place **but we also haven't given the option to delete it. So, even if it was selected accidentally user was not able to delete it.** 

Activities on account level are linked to activities on Site level and on deleting it from site accounts were not able to link it's existing activities.

**As a resolution I have removed all the activities from SAhosted accounts. We might get the same issues with other non vcc sites if there were any vcc activity selected on account level.**
